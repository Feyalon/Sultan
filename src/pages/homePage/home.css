.Card-menu-content{
    display: grid;
    grid-template-columns: repeat(3, 250px);
    justify-content: center;
    height: auto;
    grid-template-rows: 700px;
}
.Card-menu{
    display: flex;
    flex-direction: column;
}
.content{
    margin-top: 40px;

}
.content-main{
    display: grid;
    grid-template-columns: repeat(2, 600px);
    margin-top: 60px;
    justify-content: center;
    height: auto;
}
.filters{
    width: 500px;
}
.content-headers{
    margin-left: 120px;
}
.header-content{
    display: flex;
    justify-content:space-between;
    align-items: center;
}
.pagination{
    display: flex;
    width: 100%;
    justify-content: space-between;
    align-items: center;
    height: 120px;
}
.pagination button{
    width: 90px;
    height: 45px;
    background-color: #ffed28;
    border: none;
    color: white;
    cursor: pointer;
}
.pagination button:hover{
    background-color: #a19618;
}
.pagination p{
    font-size: 20px;
}
.mobile-header{
    display: none;
}
.mobile-header-top-content{
    display: flex;
    width: 100%;
    justify-content: space-between;
    align-items: center;
    border-bottom: 1px solid #3F4E65;
}
.mobile-header-top-content button{
    height: 32px;
    width: 32px;
    border-radius: 100%;
    border: none;
    background-color: #ffed28;
    display: flex;
    align-items: center;
    justify-content: center;


    
}
.mobile-header-top-content button span{
    color: white;
}
.mobile-header-top-content .logo{
}
.mobile-header-middle-content{
    display: flex;
    align-items: center;
    width: 100%;
    justify-content: space-between;
    border-bottom: 1px solid #3F4E65;
    
}
.mobile-header-down-content{
}
.mobile-catalog{
    display: flex;
    width: 100%;
    justify-content: center;
    align-items: center;
}
h1{
    font-size: 20px;
}
.openParameters{
    background-color: #FFCA65;
    width: 30px;
    height: 30px;
    border-radius: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
}
.mobile-header-top-content{
    display: flex;
}
.mobile-header-top-content h2{
    font-size: 32px;
}