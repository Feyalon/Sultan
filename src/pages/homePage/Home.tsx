import Card from '../../components/card/Card'
import { Categories } from './../../components/categories/Categories';
import {useState, useEffect} from 'react';
import "./home.css";
import { Filters } from './../../components/filters/Filters';
import { Sorted } from './../../components/sorted/Sorted';
import { Root } from './../../components/root/Root';
import { filterList } from './../../data/filterlist';
import axios from 'axios'
import { Parameters } from './../../components/parameters/Parameters';

export function Home(){
    const [items, setitems] = useState([]);
    const [filters, setfilters] = useState([]);
    const [categoryId, setCategoryId] = useState(0);
    const [sortType, setSortType] = useState(0)
    const [orderType, setOrderType] = useState(0)
    const [editType, setEditType] = useState(false)
    const [paginationType, setPagination] = useState(1)
    const [paginationBool, setPaginationBool] = useState(true)
    const [parametersValue, setParametersValue] = useState("")    
    const [downParametersValue, setDownParametersValue] = useState("")
    const [open, setOpen] = useState(false)
    let backUp: any[] = []
    const pagination = [1,2,3,4,5,6,7,8]

    useEffect(() => {
        if(categoryId === 0){
            fetch(`http://localhost:4000/products/?_sort=${sortType > 0 ? 'price': 'name'}&_order=${orderType > 0 ? 'desc': 'asc'}&_page=${paginationType}&_limit=6&${parametersValue}&${downParametersValue}`)
                .then((response) =>  response.json())
                .then((json) => {
                    setitems(json)
                    setPaginationBool(true)
                });
        }else{
            fetch("http://localhost:4000/products/?category=" + categoryId + `&_sort=${sortType > 0 ? 'price': 'name'}&_order=${orderType > 0 ? 'desc': 'asc'}&${parametersValue}&${downParametersValue}`)
                .then((response) =>  response.json())
                .then((json) => {
                    setitems(json)
                    setPaginationBool(false)
                });
        }


    }, [categoryId, sortType, orderType, paginationType, parametersValue, downParametersValue])
    let limit = 24 / 3
    useEffect(() => {
        if(backUp.length === 0){
            localStorage.setItem('items', JSON.stringify(items))
        }else{
            localStorage.setItem('items', JSON.stringify(backUp))
        }
        
    }, [backUp])
    const data_map = JSON.parse(localStorage.getItem('items') || '{}')
    console.log(data_map.length)

    const remove = (id: any) => {
        axios.delete(`http://localhost:4000/products/${id}`).then(response => console.log(response)).catch((e) => console.log(e))
    }
    function previousPage(): void {
        if(paginationType === 0){
            setPagination(1)
        }else{
            setPagination(paginationType - 1)
        }
    }

    function nextPage(): void {
        if(paginationType === 4){
            setPagination(1)
        }else{
            setPagination(paginationType + 1)
        }
    }
    return(
        <div className="content" data-testid="PostContent">
            <div className="mobile-header">
                
                <div className="mobile-header-down-content">
                    <div className="mobile-header-top-content" onClick={() => setOpen(!open)}>
                        <h2>
                            Подбор по параметрам
                        </h2>
                        <span className="openParameters">
                            {!open &&
                            <svg width="10" height="6" viewBox="0 0 10 6" fill="none" xmlns="http://www.w3.org/2000/svg">
                            <path d="M1 1L5 5L9 1" stroke="#3F4E65"/>
                            </svg>
                            }
                            {open &&
                                <svg width="10" height="6" viewBox="0 0 10 6" fill="none" xmlns="http://www.w3.org/2000/svg">
                                <path d="M9 5L5 1L1 5" stroke="#3F4E65"/>
                                </svg>
                                
                            }
                        </span>
                    </div>
                    {open && <Parameters value={parametersValue} onChangeValue={(i) =>setParametersValue(i)} onDownChangeValue={(e) => setDownParametersValue(e)} />}
                    <Categories value={categoryId} onChangeCategory={(i) => setCategoryId(i)} />
                    <Sorted value={sortType} onChangeSort={(i) => setSortType(i)} onChangeOrder={(i) => setOrderType(i)} />
                </div>
            </div>

            <div className="content-headers">
                <div className="header-content">
                    <Root />
                    <Sorted value={sortType} onChangeSort={(i) => setSortType(i)} onChangeOrder={(i) => setOrderType(i)} />
                </div>
                <Categories value={categoryId} onChangeCategory={(i) => setCategoryId(i)} />
            </div>
            
            <div className="content-main">
                
                <div className="filters">
                    <h2>
                        Подбор по параметрам
                    </h2>
                   
                    <Parameters value={parametersValue} onChangeValue={(i) =>setParametersValue(i)} onDownChangeValue={(e) => setDownParametersValue(e)} />
                    {filterList.map(filter => <Filters filter={filter}/>)}
                </div>
                
                <div className="Card-menu">
                    
                    <div className="Card-menu-content" data-testid="cards">
                        {items.map(product => <Card product={product} onDeleteClick={(i) => remove(i)} />)}
                    </div>
                    {paginationBool &&
                    <div className="pagination">
                        <button onClick={previousPage}>
                            Предыдущий
                        </button>
                        <p>
                            {paginationType === 0? paginationType + 1 : paginationType}
                        </p>
                        <button onClick={nextPage}>
                            Следующий 
                        </button>
                        
                    </div>
                 }
                </div>
                
            </div>
            
        </div>
    )   
}