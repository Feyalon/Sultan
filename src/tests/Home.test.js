import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { Home } from './../pages/homePage/Home';

describe('Home', () => {
    it('Render Home', () => {
        const {renderHome} = render(<Home />);
        expect(renderHome).toMatchSnapshot();
        
    });
    
    it('Click-pagination', () => {
        render(<Home />); 
        const btn = screen.getByText('Следующий')
        fireEvent.click(btn)
        
    });
});
