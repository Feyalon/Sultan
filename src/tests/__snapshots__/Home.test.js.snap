// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Render Home 1`] = `undefined`;
